src/main/java/com/google/sps/servlets/AddExperienceServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Tutor;
+import com.google.sps.data.TimeRange;
+import com.google.sps.data.TutorSession;
+import com.google.sps.data.SampleData;
+import com.google.sps.data.Experience;
+import com.google.sps.utilities.ExperienceDatastoreService;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Calendar;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-experience")
+public class AddExperienceServlet extends HttpServlet {
+    private ExperienceDatastoreService datastore;
+
+    public void init() {
+        datastore = new ExperienceDatastoreService();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //Set default value to -1 
+        String studentID = Optional.ofNullable((String)request.getSession(false).getAttribute("userId")).orElse("-1");
+        String experience = request.getParameter("experience");
+
+        if(studentID.equals("-1")) {
+            response.setContentType("application/json");
+            response.getWriter().println("{\"error\": \"There was an error adding experience.\"}");
+            return;
+        }
+
+        Experience newExperience = new Experience(studentID, experience);
```

Comment:
Can you fill me in on experiences? I think I missed when this was added.

Reply:
I added experiences and goals last week to the progress tracker feature. Experiences keep track of any previous experiences a user might have had on the topics they are studying (past classes, courses, certificates,..) and goals keep track of goals the student sets for themselves. These are displayed to the tutor so that they can prepare a tutoring sessions that suits the student.